File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    for i in range(len(sizes)):
        if sizes[i][0] < sizes[i][1]:
            sizes[i] = [sizes[i][1], sizes[i][0]]
    w_max = max(sizes, key=lambda x:x[0])[0]
    h_max = max(sizes, key=lambda x:x[1])[1]
    return w_max * h_max

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        // 가로 >= 세로가 되게 돌리고 각각 최댓값 구하기
        int maxWidth = 0;
        int maxHeight = 0;
        for (int[] size : sizes) {
            int w, h;
            if (size[0] >= size[1]) {
                w = size[0];
                h = size[1];
            } else {
                w = size[1];
                h = size[0];
            }
            maxWidth = w >= maxWidth ? w : maxWidth;
            maxHeight = h >= maxHeight ? h : maxHeight;
        }
        return maxWidth * maxHeight;
    }
}